src/Castle.Facilities.AspNetCore/CastleWindsorRegistrationExtensions.cs

```diff
@@ -0,0 +1,102 @@
+		/// <param name="container">Windsor container which activators call resolve against</param>
+		public static void AddCastleWindsor(this IServiceCollection services, IWindsorContainer container)
+		{
+			services.AddRequestScopingMiddleware(container.BeginScope);
+			services.AddCustomControllerActivation(container.Resolve);
+			services.AddCustomTagHelperActivation(container.Resolve);
+			services.AddCustomViewComponentActivation(container.Resolve);
+			container.Kernel.Resolver.AddSubResolver(new LoggerDependencyResolver(services));
+			container.Kernel.Resolver.AddSubResolver(new FrameworkConfigurationDependencyResolver(services));
+		}
+
+		/// <summary>
+		/// Use this to register all Controllers, ViewComponents and TagHelpers
+		/// </summary>
+		/// <typeparam name="TTypeAssembly">Type from assembly to scan and register ASP.NET Core components</typeparam>
+		/// <param name="app">Application builder retained as extension</param>
+		/// <param name="container">Windsor container to register framework types in</param>
+		public static void UseCastleWindsor<TTypeAssembly>(this IApplicationBuilder app, IWindsorContainer container)
+		{
+			container.Register(Classes.FromAssemblyInThisApplication(typeof(TTypeAssembly).Assembly).BasedOn<Controller>().LifestyleScoped());
+			container.Register(Classes.FromAssemblyInThisApplication(typeof(TTypeAssembly).Assembly).BasedOn<ViewComponent>().LifestyleTransient());
+			container.Register(Classes.FromAssemblyInThisApplication(typeof(TTypeAssembly).Assembly).BasedOn<TagHelper>().LifestyleTransient());
+		}
+
+		/// <summary>
+		/// For registering middleware that consumes services in the constructor known to Castle Windsor. You can use 
+		/// conventional methods for registering your middleware but then you have to re-register your dependencies
+		/// in the ASP.NET IServiceCollection. You should avoid doing this if possible and use this extension instead. 
+		/// </summary>
+		/// <typeparam name="T">Type of service that implements Microsoft.AspNetCore.Http.IMiddleware</typeparam>
+		/// <param name="app">Application builder</param>
+		/// <param name="container">Windsor container</param>
+		public static void UseCastleWindsorMiddleware<T>(this IApplicationBuilder app, IWindsorContainer container) where T : class, IMiddleware
+		{
+			container.Register(Component.For<T>());
+			app.Use(async (context, next) =>
+			{
+				var resolve = container.Resolve<T>();
+				await resolve.InvokeAsync(context, async (ctx) => await next());
+			});
+		}
+
+		/// <summary>
+		/// Optional helpful exception to validate against types being installed into Windsor from the 'Microsoft.AspNetCore'
+		/// namespace. This guards against 'Torn Lifestyles' and 'Captive Depenencies'.
+		/// </summary>
+		/// <param name="container">Windsor container</param>
+		public static void ValidateConfiguration(this IWindsorContainer container)
```

Comment:
@fir3pho3nixx I just realised this is an extension method on `IWindsorContainer`, I think this will confuse users when they import the `Castle.Facilities.AspNetCore` namespace as the usage will appear:

```
container.ValidateConfiguration();
```

And it doesn't validate configuration for the whole container, just for this facility.

I'm going to merge this pull request, but I think it would be a worthwhile fix to scope this either by changing the method name or something else.

Reply:
I will slot this into a separate PR. How about renaming it to ValidateNoFrameworkComponentsAreRegistered?